syntax = "proto3";

package reg;

service Registry {
    rpc map (MapRequest) returns (MapReply) {}
    rpc unmap (UnMapRequest) returns (UnMapReply) {}
    rpc query (QueryRequest) returns (QueryReply) {}
    rpc finalize (FinalizeRequest) returns (FinalizeReply) {}
}

message MapRequest {
    string service_name = 1;
    string pm_address = 2;
}

message MapReply {
    int32 num_services = 1;
}

message UnMapRequest {
    string service_name = 1;
}

message UnMapReply {
    int32 success = 1;
}

message QueryRequest {
    string service_name = 1;
}

message QueryReply {
    string pm_address = 1;
}

message FinalizeRequest {
}

message FinalizeReply {
    int32 num_services = 1;
}